src/ImageSharp/Formats/Png/PngDecoderCore.cs

```diff
@@ -637,20 +672,30 @@ private void ProcessDefilteredScanline<TPixel>(byte[] defilteredScanline, int ro
 
                 case PngColorType.RgbWithAlpha:
 
-                    for (int x = 0; x < this.header.Width; x++)
-                    {
-                        int offset = 1 + (x * this.bytesPerPixel);
+                    this.RgbWithAlpha(defilteredScanline, pixels);
 
-                        byte r = defilteredScanline[offset];
-                        byte g = defilteredScanline[offset + this.bytesPerSample];
-                        byte b = defilteredScanline[offset + (2 * this.bytesPerSample)];
-                        byte a = defilteredScanline[offset + (3 * this.bytesPerSample)];
+                    break;
+            }
+        }
 
-                        color.PackFromBytes(r, g, b, a);
-                        pixels[x, row] = color;
-                    }
+        /// <summary>
+        /// Processing the RGB with Alpha is a straight copy
+        /// </summary>
+        /// <typeparam name="TPixel">The type of pixel</typeparam>
+        /// <param name="defilteredScanline">The completed scanline</param>
+        /// <param name="pixels">The pixel accessor</param>
+        private unsafe void RgbWithAlpha<TPixel>(byte[] defilteredScanline, PixelAccessor<TPixel> pixels)
+            where TPixel : struct, IPixel<TPixel>
+        {
+            int offset = this.bytesPerSample * 4;
+            int width = this.header.Width * offset;
+            int pixelId = this.currentRow * this.header.Width;
+            Rgba32[] pixelArray = pixels.PixelArray as Rgba32[];
```

Comment:
This works only with `Image<Rgba32>`! (Our fault the non-rgba case is not covered by tests though.)

A general and simple way to do this conversion:
`BulkPixelOperations<TPixel>.Instance.PackFromXyzwBytes(sourceColorSpan, destByteSpan, count)`
In case of `Rgba32` it will delegate to an [optimized memory copy call](https://github.com/JimBobSquarePants/ImageSharp/blob/master/src/ImageSharp/Common/Memory/BufferSpan.cs#L25).